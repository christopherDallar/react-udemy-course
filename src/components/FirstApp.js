import React from 'react';
import PropTypes from 'prop-types';

// props means properties

const FirstApp = ({ regard, subtitle }) => {
  return (
    <>
      <h1>{regard}</h1>
      <p>Mi primera app</p>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  regard: PropTypes.string.isRequired,
  subtitle: PropTypes.any,
  // other: PropTypes.number.isRequired
};
FirstApp.defaultProps = {
  subtitle: '',
  // subtitle: 'Soy un subtitle',
};

export default FirstApp;
