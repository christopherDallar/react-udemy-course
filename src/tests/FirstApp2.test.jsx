import React from 'react';
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import FirstApp from '../components/FirstApp';
import {
  getAllByText,
  render,
  screen,
} from '@testing-library/react';

describe('Test on FirstApp Component', () => {
  const regard = 'Hello, I am Goku';
  const subtitle = 'I am a subtitle';

  test('must to match with snapshot', () => {
    const { container } = render(
      <FirstApp regard={regard} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('must to show Hello, I am Goku message', () => {
    render(<FirstApp regard={regard} />);

    // screen.debug()
    expect(screen.getByText(regard)).toBeTruthy();
  });

  test('must to show title on h1', () => {
    render(<FirstApp regard={regard} />);

    expect(
      screen.getByRole('heading', { level: 1 }).innerHTML,
    ).toContain(regard);
  });

  test('must to show subtitle passed by props', () => {
    render(
      <FirstApp regard={regard} subtitle={subtitle} />,
    );

    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
