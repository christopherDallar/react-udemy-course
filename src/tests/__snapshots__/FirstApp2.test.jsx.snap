// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on FirstApp Component must to match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hello, I am Goku
     
  </h1>
  <p>
    Mi primera app
  </p>
  <p
    class="subtitle"
  >
    I am subtitle
  </p>
  <p
    class="subtitle"
  >
    I am subtitle
  </p>
</div>
`;
